### ISSUE TITLE
[Face attributes Model] Output of sunglasses probability is always 1

### ISSUE THREAD

**Issue by chouxscream** [quic/ai-hub-models#179]
Hello, I have question on face attributes model output.

While other outputs seems to be correct after using softmax function, output of sunglasses is always equal to or almost 1.0.
I have not used softmax to sunglasses, as pointed out in https://github.com/quic/ai-hub-models/issues/137.
Regardless of input images with or without sunglasses on, the output does not represent sunglasses correctly.
Am I missing something, such as preprocessing input data? Below is the command that I have used.

python -m qai_hub_models.models.face_attrib_net.demo --image {image_path}

Thank you in advance.

**Comment by mestrona-3** [comment]
Hi @chouxscream, thanks for the question! I've asked the team that worked on this model and they have some follow up questions:

Yes, you are handling the sunglasses result correctly, and there's no need to perform softmax. Though we'll need more information to pinpoint the exact problem. If you could provide some failure cases, it would be easier for us to identify the root cause. Are you using RGB data?

Regarding preprocessing, for optimal results, we perform an affine transformation to the images and crop the faces according to the following criteria: center of eyes and mouth are supposed to be well aligned horizontally and the distance from the top of the image to the eyes, from the eyes to the mouth, and from the mouth to the bottom should all equal to 1/3 of the image height.

**Comment by chouxscream** [comment]
Yes, I used RGB data.

This is the output using sample face image from qualcomm : 
![Image](https://github.com/user-attachments/assets/e7c364c6-aa10-4121-af7a-bdf02b5e0df4)
"sunglasses": [
        0.9999938011169434,
        6.184478934301296e-06
    ]

This is the output using sunglass face image downloaded from internet : 
![Image](https://github.com/user-attachments/assets/b9362763-8e82-4403-9127-250ceeb5ee81)
"sunglasses": [
        0.99959796667099,
        0.0004019975021947175
    ]

I have uesd SCRFD detector for cropping face from input image, so should I use other face detection method before using attribute model?

**Comment by mestrona-3** [comment]
Hi chouxscream, thank you for the additional details. You're right it does appear that the results are incorrect and significantly differ from our previous runs, both online and offline. We'll need more information to dig into the root cause. 

Could you provide us with the following:

- which device are you deploying to
- can you provide some unit test data with groundtruth for us to investigate the model use with proper preprocessing/post processing

In the meantime, we'd recommend trying the quantized version as the accuracy should improve, depending on the platform you're deploying the model on. We also wanted to share that we'll soon be updating a new version trained using RGB instead of NIR data, so you should expect better accuracy in the near future release.

**Comment by chouxscream** [comment]
> which device are you deploying to

I plan to export onnx model to other NPU device and since quantization is applied separately, I did not use quantized version from qualcomm.

> We also wanted to share that we'll soon be updating a new version trained using RGB instead of NIR data,

So should I use NIR image input instead of RGB?

> can you provide some unit test data with groundtruth for us to investigate the model use with proper preprocessing/post processing

For preprocessing, crop face bounding box from scrfd-10g detector and  resize to (128,128)
No postprocessing is applied after inference.

**Comment by zxylnnu** [comment]
I think I meet the same problem with you. I download the published onnx format model, which shows the input is (1, 3, 128, 128). I test it with RGB images, but the output always cant shows a right answer.

**Comment by mestrona-3** [comment]
hi @chouxscream sorry for the delay in responding. Is this with the quantized model or fp16? We've made some changes recently, that affected the quantization of this model. If you are still hitting this issue it could be that the quantization is bad, and you could try redownloading. If it's the fp16 model, do let us know and we can investigate further. Apologies!!

**Comment by kory** [comment]
Folks, thanks for the report. We investigated internally.

This model expects Infrared image grayscale input, stacked from 1 to 3 channels. The problem you're facing is you're passing an RGB image, which the model wasn't trained on. 

We will update the demo & documentation for this model in the next release (in 2 weeks) to reflect this.

If you try a Infrared grayscale image (stacked to 3 channels) and still get poor results, please let us know.
